core/controllers/reader_test.py

```diff
@@ -205,6 +248,103 @@ def test_huge_answers(self):
             'a' * 1000500, 'Sorry, nope, we didn\'t get it', '')
 
 
+class ReaderClassifyTests(test_utils.GenericTestBase):
+    """Test reader.classify using the sample explorations.
+
+    Since the end to end tests cover correct classification,
+    ReaderClassifyTests is only checking which of the hard/soft/classifier
+    rules triggered on an input.
+    """
+
+    def setUp(self):
+        super(ReaderClassifyTests, self).setUp()
+        self.init_classify_inputs('16')
+
+    def init_classify_inputs(self, exploration_id):
+        exp_services.delete_demo(exploration_id)
+        exp_services.load_demo(exploration_id)
+
+        self.EXP_ID = exploration_id
+        self.EXP_STATE = (
+            exp_services.get_exploration_by_id(exploration_id).states['Home'])
+
+    def determine_triggering_rule(self, answer):
+        response = reader.classify(self.EXP_ID, self.EXP_STATE, answer,
+            {'answer': answer})
+        return response['matched_rule_type']
+
+    def test_hard_triggers(self):
+        """All these responses trigger the hard classifier.
+
+        Note: Any response beginning with 'hardrule' is a hard rule
```

Comment:
Perhaps: *Any response beginning with 'hardrule' will result in reader.classify selecting a hard rule.*